ui/widgets/calendar.js

```diff
@@ -133,12 +125,31 @@ return $.widget( "ui.calendar", {
 		this._addClass( $( event.currentTarget ), null, "ui-state-hover" );
 	},
 
+	_select: function( event ) {
+		var oldValue = this.options.value ? this.options.value.getTime() : "";
+
+		this._setOption( "value", new Date( $( event.currentTarget ).data( "timestamp" ) ) );
```

Comment:
Could this access `this.activeDescendant` instead? Would make the ENTER key event handler simpler, since it wouldn't need to pass the event along.